packages/xforms-engine/src/parse/body/control/InputControlDefinition.ts

```diff
@@ -11,10 +12,18 @@ export class InputControlDefinition extends ControlDefinition<'input'> {
 
 	readonly type = 'input';
 	readonly appearances: InputAppearanceDefinition;
+	readonly rows: number | null;
```

Comment:
The `rows` implementation will be completed in this issue: https://github.com/getodk/web-forms/issues/301